yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala

```diff
@@ -64,7 +68,7 @@ private[spark] class YarnClientSchedulerBackend(
     // SPARK-8851: In yarn-client mode, the AM still does the credentials refresh. The driver
     // reads the credentials from HDFS, just like the executors and updates its own credentials
     // cache.
-    if (conf.contains("spark.yarn.credentials.file")) {
+    if (!conf.contains(PRINCIPAL.key) && conf.contains("spark.yarn.credentials.file")) {
```

Comment:
hi @vanzin 
the method: org.apache.spark.deploy.yarn.ExecutorDelegationTokenUpdater.updateCredentialsIfRequired() invokes the same operation of updating the current user credential and correct me if I am wrong; if user has logged in through a keytab file already then ideally there shouldn't be any need to update the credentials at regular intervals. This thread of updating the credential might be useful for executors where there is only CREDENTAILS_FILE or delegation token.